app/controllers/admin/tutorials_controller.rb

```diff
@@ -5,15 +5,11 @@ def edit
 
   def create
     if missing_params.present?
-      flash[:error] = "Please enter a #{missing_params}."
-      redirect_to new_admin_tutorial_path
+      missing_params_error
```

Comment:
created 3 private methods to reduce method length. because they are mostly flash messages and redirects they cannot be put in a model

Reply:
![](https://media2.giphy.com/media/cdr4V05plfxv2/giphy.gif?cid=5a38a5a22475195a9dd503f5c45a44fee77db970ae7ec0a9&rid=giphy.gif)